File: AlgoDS/BinarySearch.cpp
#include <iostream>

using namespace std;

int binarySearch(int arr[], int left, int right, int number)
{
    while(left <= right)
    {
        int mid = (left+right)/2;

        if(arr[mid] == number){
            return mid;
        }

        //if value on mid is less than number to be search
        if(arr[mid] < number){
            left = mid +1;
        }else{
            right = mid -1;
        }

    }
    return -1;
}

int main(void)
{
    int arr[] = {1,2,3,5,88,100};
    int arrSize = sizeof(arr)/sizeof(arr[0]);
    int numberToSearch = 0;
    cin >> numberToSearch;
    int res = binarySearch(arr, 0, arrSize, numberToSearch);
    cout << "number " << numberToSearch << " is located at arr : " << res << " ";
}

File: AlgoDS/src/BinarySearch.java
class BinarySearch {
	
	int BinarySearch(int arr[], int left, int right, int value)
	{
		while(left <= right)
		{
			int mid = (left + right)/2;
			
			if(arr[mid] == value){
				return mid;
			}
			
			if(arr[mid] < value){
				left = mid + 1; 
			}else{
				right = mid - 1;
			}
		}
		
		return -1;
	}
	

	public static void main(String[] args) {
		BinarySearch object = new BinarySearch();
		int arr[] = {1,2,3,7,9,100};
		int arrLength = arr.length-1;
		int value = 7;
		
		int result = object.BinarySearch(arr, 0, arrLength, value);
		if(result == -1){
			System.out.println("Value not found!");
		}else{
			System.out.println("Value found on index : " + result);
		}
	}
}
